File: C++/ArrayString/01_LongPressedName.cpp
#include <bits/stdc++.h>
using namespace std;

bool isLongPressedName(string name, string typed)
{
    int i = 0, j = 0;
    if (typed.size() < name.size())
        return false;

    while (i < name.size() || j < typed.size())
    {
        // this is for the condition when characters are same
        //  we increment the i and j
        if (i < name.size() and j < typed.size() and name[i] == typed[j])
        {
            i++;
            j++;
        }
        // if not equal then there is possibility that typed[j] can be equal to i-1
        else if (i > 0 and j < typed.size() and name[i - 1] == typed[j])
        {
            j++;
        }
        else
            return false;
    }
    //cout << i << " " << j << endl;
    //         if(i<name.size()) return false;
    //         if(j<typed.size()){

    //         }
    return true;
}

File: Python/ArrayString/01_LongPressedName.py
def isLongPressedName(self, name: str, typed: str) -> bool:
    i = 0
    j = 0
    if (len(typed) < len(name)):
        return False

    while (i < len(name) or j < len(typed)):
        ''' this is for the condition when characters are same
            we increment the i and j '''
            
        if (i < len(name) and j < len(typed) and name[i] == typed[j]):
            i += 1
            j += 1

        # if not equal then there is possibility that typed[j] can be equal to i-1
        elif (i > 0 and j < len(typed) and name[i - 1] == typed[j]):
            j += 1

        else:
            return False

    return True